tests/typedarray.rs

```diff
@@ -45,5 +45,9 @@ fn typedarray() {
 
         typedarray!(in(cx) let array: Uint32Array = rval.get());
         assert_eq!(array.unwrap().as_slice(), &[1, 3, 5, 0, 0][..]);
+
+        typedarray!(in(cx) let mut array: Uint32Array = rval.get());
+        array.as_mut().unwrap().update(Some(&[0, 2, 4, 6]));
+        assert_eq!(array.unwrap().as_slice(), &[0, 2, 4, 6, 0][..]);
     }
```

Comment:
Let's add another test in this file that uses `#[should_panic]` that calls update with an argument that is longer than the typed array that it is replacing, too.